CMakeLists.txt

```diff
@@ -0,0 +1,26 @@
+# Copyright 2018 Mike Dev
+# Distributed under the Boost Software License, Version 1.0.
+# See accompanying file LICENSE_1_0.txt or copy at https://www.boost.org/LICENSE_1_0.txt
+#
+# NOTE: CMake support for Boost.Mpl is currently experimental at best
+#       and the interface is likely to change in the future
+
+cmake_minimum_required( VERSION 3.5 )
+project( BoostMpl LANGUAGES CXX )
+
+add_library( boost_mpl INTERFACE )
+add_library( Boost::mpl ALIAS boost_mpl )
+
+target_include_directories( boost_mpl INTERFACE include )
+
+target_link_libraries( boost_mpl
```

Comment:
How do we keep this list updated?  Can the b2 build fail if this list is not equal to the actual primary dependencies?

Reply:
I don't see the list of dependencies being static as acceptable.  They change all the time, but I don't see a solution here.

Reply:
>  They change all the time, 

Really? Is there so much active development in boost.mpl? Keep in mind that this file only specifies direct depencies.

Personally, I think adding dependencies to a library willy nilly is an absolute no-go anyway.

Reply:
What if someone checks in a small bugfix to mpl and I commit it, and it pulls in a new header and a new library, which pulls in three others... a static list is not maintainable.  Whether or not the repository is under active development or not is another matter.  This pattern will be copied into repositories that are active, and then it will not be maintainable.

Reply:
@jeking3: Sorry for not answering earlier, but life happened.

> What if someone checks in a small bugfix to mpl and I commit it, and it pulls in a new header and a new library

I can just repeat myself: A small bugfix should not change the list of dependencies. 
Also, it is not as if this would not be detectable: E.g. through build tests, manual/automatic comparison of the output of boostdep with the cmake file or simply by checking the patch for new `#include` statements.

> which pulls in three others...

This file is only concerned with the direct dependencies not the transitive ones

> a static list is not maintainable

All the projects I've been working on have used static lists of dependencies in one form or another (If only a static list of include directories and binaries to link with) and if I'm not mistaken, so do compiled boost libraries in their jamfiles, the other boost libraries that have gained cmake files so far and an best I can tell, full fledged solutions like bcm do this too (e.g. https://github.com/boost-cmake/boost/blob/master/libs/mpl/CMakeLists.txt). 

So I really don't agree with that statement and actual think that a static list of dependencies is the norm rather than the exception.

However, in the end, the only thing that matters is what you are willing to merge into your library. So what specifically is needed to make this acceptable for you? 

>  This pattern will be copied into repositories that are active, and then it will not be maintainable.

I can understand that manual maintainence is a problem for CMT libraries, but for a repository under active development, updating the cmake file every now and then should really be the least  of the problems (It seems to me that the newest / most actively maintained libraries use cmake for the develoment anyway).